### DISCUSSION TITLE
Cleanup /api/v1/info

### DISCUSSION THREAD

**Discussion by MrZammler** [netdata/netdata#12365] [Ideas]
When the analytics project started, it was decided to use /api/v1/info as a place from where dashboard could take attributes and include them in it's analytics messages.

Some of the attributes we have there might make sense for an informational endpoint (e.g. memory mode), some not really (e.g. dashboard-used). What makes it more confusing is that some of these attributes will be populated only if analytics is running.

So do you agree that we perhaps need to:

1) If we need some of those there, detach them from the analytics code so that they're always be there.
2) If don't want some of them either remove them (does the dashboard need to send them?) or move them to their own endpoint specifically for this purpose.

**Comment by MrZammler** [comment]
Tagging @andrewm4894 and @jacekkolasa

**Comment by underhood** [comment]
Well I liked idea from @Ferroin which is new endpoint `/api/v1/anonymous-analytics` (or whatever you want to name it) which will show exactly what is sent to cloud.

Current `/api/v1/info` thing is broken for multiple reasons:
- it will show `null` for keys like `memory-mode` in case when `DO_NOT_TRACK` is enabled, this is terrible design as this is just config value which is available regardless of anonymous statistics (examples like this are countless)
- things are capped in places like `dashboard-used` which is forced to max at 255 -> this will be confusing for user (he will refresh dashboard and counter will not increase)

In general `/api/v1/info` HAS to be independent on analytics and show correct data always (from source).
If it is desired to show exact data analytics are sending to the posthog this should be done with another dedicated endpoint (show raw data sent). This is good also for users trust (we can point them from documentation -> if you want to know what is being sent look at `/api/v1/whatissent`

FYI, I will be doing changes required probably as part of https://github.com/netdata/netdata/pull/11603 which is issue that led me to discover these and other problems (which i have to fix to proceed with this PR).

**Comment by andrewm4894** [comment]
we had mentioned as some stage making `/api/v1/info` also a bit more dynamic potentially such that you could 'ask' it about different things as opposed to just the wall of text today. 

but agree that its trying to maybe do too many things now at this stage.

there is some merit in just needing to go to `api/v1/info` for info about the node. but maybe more separation will just be cleaner longer run perhaps.

**Reply by underhood** [reply]
if we insist on keeping it in `info` we have to separate it to `analytics` subgroup as those values are null when analytics are disabled. Some of the keys are representing config files key.... then it is broken for users (absolutely cannot stay as it is now... it is a bug).
